core/manager: Deprecate StartAuxiliaryScope() method

Werkov opened the pull request:
The method was added with migration of resources in mind (e.g. process's allocated memory will follow it to the new scope), however, such a resource migration is not in cgroup semantics. The method may thus have the intended users and others could be guided to StartTransientUnit().

Since this API was advertised in a regular release, start the removal with a deprecation message to callers.
Eventually, the goal is to remove the method to clean up DBus API and simplify code (removal of cgroup_context_copy()).

@msekletar I remember you considered the removal. What do you think about this deprecation now?

YHNdnzj commented:
Hmm, I remember some intensive discussion over the use case and this new functionality. Are all those no longer relevant?

YHNdnzj commented:
(I mean, I didn't like this when it was introduced either, especially because of the reasons mentioned in the PR description, but it was merged anyway since people were keen on such mechanism)

poettering commented:
CI not happy btw

poettering commented on src/core/dbus-manager.c:
```diff
@@ -2914,6 +2914,7 @@ static int aux_scope_from_message(Manager *m, sd_bus_message *message, Unit **re
                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS,
                                          "Name \"%s\" of auxiliary scope doesn't have .scope suffix.", name);
 
+        log_unit_warning(from, "Deprecated StartAuxiliaryScope() call");
```
Can you make this a proper english sentence, i.e. "D-Bus call StartAuxiliaryScope() has been invoked which is deprecated. Removal is imminent."

poettering commented on src/core/dbus-manager.c:
```diff
@@ -3035,6 +3036,9 @@ static int method_start_aux_scope(sd_bus_message *message, void *userdata, sd_bu
 
+        log_once(LOG_WARNING, "StartAuxiliaryScope() is deprecated because state of resources cannot be migrated between cgroups. "
+                              "Please report this to systemd-devel@lists.freedesktop.org or "
+                              "https://github.com/systemd/systemd/issues/ if you see what unit uses this");
```
"… if you see this message and know the software making use of this functionality."

poettering replied:
could you also add the SD_BUS_VTABLE_DEPRECATED flag to the bus call in the vtable?

msekletar commented:
I don't have a strong opinion about it so I don't care either way. However, regarding the resource migration...Is the claim actually true is full? AFAICT, the only resource that can't be migrated is memory. Now, the question is whether this alone (if I am correct) is actually good enough reason to get rid of this completely.

Werkov commented:
Thanks both for the review.

Lennart:  Addressed your remarks. (Thanks especially for the `SD_BUS_VTABLE_DEPRECATED` bit, that should hopefully satisfy both me and the CI docs checker.)

Michal: It's true for "stateful" resources which is mainly memory, possibly some state that's tracked by eBPF. There may be some transitional effects wrt cpu or IO scheduling but those would be negligible (it's not that much of a state). I toned down the wording (no _imminent_ removal) so that this serves as survey of actual users. (For me the lack of migration is not the reason but lack of users is.)
